repo_policy_compliance/check.py

```diff
@@ -164,28 +169,28 @@ def collaborators(github_client: Github, repository_name: str) -> Report:
     return Report(result=Result.PASS, reason=None)
 
 
-def _branch_external_fork(
-    repository_name: str, source_repository_name: str, push_logins: set[str]
-) -> bool:
+def _branch_external_fork(repository: Repository, source_repository_name: str) -> bool:
     """Check whether a branch is an external fork.
 
     A external fork is a fork that is not owned by a user who has push or above permission on the
     repository.
 
     Args:
-        repository_name: The name of the repository to run the check on.
+        repository: The repository to run the check on.
         source_repository_name: The name of the repository that contains the source branch.
-        push_logins: The logins from users with push or above permission or above on the
-            repository.
 
     Returns:
         Whether the branch is from a external fork.
     """
-    if repository_name == source_repository_name:
+    if repository.full_name == source_repository_name:
         return False
 
-    # Check if the owner of the fork also has push or higher permission
-    if source_repository_name.split("/")[0] in push_logins:
+    fork_username = source_repository_name.split("/")[0]
+
+    # Check if owner of the fork already has push or higher permission (not an external user)
+    # This additional check is a fail-safe to paginated results of push_logins.
```

Comment:
Can the reference to `push_logins` be removed now?

Reply:
Removed, thank you!